### PR TITLE
Term changes must be persisted to disk.

### PR DESCRIPTION
**Pull request by cannium**

### CODE REVIEW

**Comment by otoolep** [comment]
Thanks @cannium -- good catch.

Reviewed by @benbjohnson -- merging on green. I had to kick the build due to a known issue with Continuous Queries.

File: raft/log.go [hunk 1]
```diff
@@ -959,7 +959,7 @@ func (l *Log) candidateLoop(closing <-chan struct{}) State {
 
 	// Increment term and request votes.
 	l.mu.Lock()
-	l.term++
+	l.mustSetTerm(l.term+1)
 	l.votedFor = l.id
 	term := l.term
```
Comment (otoolep) [inline raft/log.go]: This is not actually quite right, since `l.term` is no longer being incremented.

Comment (otoolep) [inline raft/log.go]: Actually, `mustSetTerm` takes care of setting `l.term` so we're actually OK.

**Comment by otoolep** [comment]
Closed in favour of https://github.com/influxdb/influxdb/pull/2219 as I need to make more changes to the branch.

**Comment by otoolep** [comment]
@cannium -- can you also sign the CLA? You can find it here:

http://influxdb.com/community/cla.html

**Comment by cannium** [comment]
Signed. Just curious, what's the difference between this patch and that of  #2219 ?

**Comment by otoolep** [comment]
The CHANGELOG. I needed this change in quickly to meet a build requirement.

**Pull request closed without merging**